### DISCUSSION TITLE
hide API key ?

### DISCUSSION THREAD

**Discussion by knupel** [gatsbyjs/gatsby#37457] [Help] [answered]
Hello, I work on API google map with the https://github.com/googlemaps/react-wrapper.  The only solution I found to pass the Api key  from the the `.env` `MY_KEY` it's with the prefix `GATSBY` like that `GATSBY_MY_KEY`. But if I'm understanding well the process, this solution give the opportunity to catch the `KEY`by a malicious people.
I try to find another solution to pass the `API_KEY`and keep this one secret but I don't find any solution,  except maybe to pass by a creation of plugin, but it's too hard for me.
So if there is the simplest solution... I'm very curious to know it.
here m y code about this project : https://github.com/knupel/gatsby_react_lab/blob/main/src/pages/api/api_google_map_react.tsx

and Happy New Year !!!

**Comment by knupel** [comment, marked as answer]
This discussion can be close, because I'm a newbie. I just realize the Google API ask for the URL of the site need to the access to it, so  I suppose there is no problem is the API key is stolen !

**Reply by violy** [reply]
Yes ! the Google Maps API key is public ! but Google provide domain-based security.
